### ISSUE TITLE
listen unix ./.overmind.sock: bind: operation not supported

### ISSUE THREAD

**Issue by dudintv** [DarthSim/overmind#88]
I'm trying to setup Overmind on the WSL  (Linux inside Windows) Ubuntu 16.04.
And, after this command:

`root@big-pc:/mnt/c/rails/vizmarket# overmind s`

 I've got an error:

```
system    | Tmux socket name: overmind-vizmarket-f2ZGVMrcSIdyY0QKfLB0X_
system    | Tmux session ID: vizmarket
system    | Listening at ./.overmind.sock
overmind: listen unix ./.overmind.sock: bind: operation not supported
```

What should I do to make it works?

**Comment by DarthSim** [comment]
Hey!
Looks like Unix sockets don't work properly on WSL. I'll check this later.

As a workaround, I added [ability to bind to TCP network instead of Unix socket](https://github.com/DarthSim/overmind#using-tcp-network) to 2.2.0

**Comment by Pagehey** [comment]
Hey @dudintv !

**Comment by dudintv** [comment]
@Pagehey
I've abandoned that and moved to my mac book. WSL provides too many shits. It could look like ok but at some moment it crashes unexpectedly.

**Comment by tmbs** [comment]
Same issue on macOS 11.2.3 with Docker 3.3.0 running `sam: sam local start-api --warm-containers=EAGER --env-vars=env.json`, works fine without `--warm-containers=EAGER`.

**Comment by DarthSim** [comment]
@tmbs do you run sam with Overmind or Overmind with sam?

**Comment by tmbs** [comment]
Current `Procfile` ran with `overmind s`:
```
sam: sam local start-api --env-vars=env.json
tsc: npm run watch
```

**Comment by DarthSim** [comment]
It is weird then as Overmind binds to the socket before running the processes, so processes shouldn't affect the binding.

**Comment by dreamalligator** [comment]
> Hey!
> Looks like Unix sockets don't work properly on WSL. I'll check this later.
> 
> As a workaround, I added [ability to bind to TCP network instead of Unix socket](https://github.com/DarthSim/overmind#using-tcp-network) to 2.2.0

Can confirm that this is still an issue in 2.2.2.

```bash
overmind start
# system | Tmux socket name: overmind-[...]
# system | Tmux session ID: [...]
# system | Listening at ./.overmind.sock
# overmind: it looks like Overmind is already running. If it's not, remove ./.overmind.sock and try again
```

```bash
overmind --version
# Overmind version 2.2.2
```

**Comment by DarthSim** [comment]
@dreamalligator This is WSL issue, so it can't be fixed inside Overmind. However, you're getting the error that tells you that you have already started an Overmind instance in the current dir. That's strange.

**Comment by dreamalligator** [comment]
Thanks @DarthSim. From turning computer off, I assume there is no cleanup step when not manually closing Overmind. Maybe I can add a script when my computer turns on to remove it, or whatever is best practice.

**Comment by ibrahima** [comment]
Sorry to bump an old thread, can make a new one if necessary. My coworker ran into this error while using Overmind inside Docker Desktop on macOS. I don't understand why this would be the case... any thoughts? We checked `uname -a` and it showed a Linux kernel 6.26. I would expect even if it was running on the macOS host Unix sockets should still be available. This seems pretty bizarre to me. We got it working with the TCP socket option but it feels strange to have to do this. Any pointers would be greatly appreciated!
